src/redux/slices/gameSlice.ts

```diff
@@ -0,0 +1,133 @@
+import { createSlice, createSelector, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from '../store';
+
+import { LayerProps, Phase } from '../../../typings/gameTypes';
+
+type Game = {
+  phase: Phase;
+  score: number;
+  enemiesCount: number;
+  enemiesGenerated: number;
+  enemiesDestroyed: number;
+  activeShortcut: number;
+  asteroids: LayerProps[];
+  bullets: LayerProps[];
+};
+
+const initialState: Game = {
+  phase: Phase.loading,
+  score: 0,
+  enemiesCount: 0,
+  enemiesGenerated: 0,
+  enemiesDestroyed: 0,
+  activeShortcut: 0,
+  asteroids: [],
+  bullets: [],
+};
+
+const gameSlice = createSlice({
+  name: 'game',
+  initialState,
+  reducers: {
+    reset: (state, action: PayloadAction<number>) => {
+      state.phase = Phase.ready;
+      state.score = 0;
+      state.enemiesCount = action.payload;
+      state.enemiesGenerated = 0;
+      state.enemiesDestroyed = 0;
+      state.bullets = [];
+      state.asteroids = [];
+    },
+    start: (state) => {
+      state.phase = Phase.playing;
+    },
+    pause: (state) => {
+      state.phase = Phase.pause;
+    },
+    gameOver: (state) => {
+      state.phase = Phase.over;
+    },
+    hit: (state, action: PayloadAction<number>) => {
+      state.score += action.payload;
+      state.enemiesDestroyed += 1;
+
+      if (state.enemiesDestroyed >= state.enemiesCount) {
+        state.phase = Phase.win;
+      }
+    },
+    out: (state) => {
```

Comment:
🧐
`out` прям очень непонятно
```suggestion
    finish: (state) => {
```

Reply:
Вообще, игровое событие **out** - это вылет астероида за пределы канваса. Из слайса **out** убрал совсем, а логику эту перенес в **removeAsteroid**. Также **hit** переименовал в **updateScore** и упростил. Теперь вроде всё должно быть понятно.